src/Runner.js

```diff
@@ -135,27 +135,38 @@ function dirFiles (dir, callback, acc) {
 
 function getAllFiles(paths, filter) {
   return Promise.all(
-    paths.map(file => new Promise(resolve => {
-      fs.lstat(file, (err, stat) => {
-        if (err) {
-          process.stderr.write('Skipping path ' + file + ' which does not exist. \n');
-          resolve([]);
-          return;
-        }
+    paths.map(
+      (file) =>
+        new Promise((resolve) => {
+          if (file.includes('*')) {
+            return fg([file])
+              .then((list) =>
+                list.filter(
+                  (name) => filter(name) && !ignores.shouldIgnore(name)
+                )
+              )
+              .then(resolve);
```

Comment:
Instead of creating a new Promise and calling `resolve` in `then`, can you return the fast-glob promise directly? For example:
```
    paths.map(
      (file) =>
        if (file.includes('*')) {
          return fg([file])
            .then((list) =>
              list.filter(
                (name) => filter(name) && !ignores.shouldIgnore(name)
              )
            );
        }

        return new Promise((resolve) => {
          // ..... the old code here
  }
);
```

Reply:
This has been processed. 

Need to test fast-glob? I have now modified to ignore files by fast-glob, ref

https://github.com/facebook/jscodeshift/pull/514/commits/e04ff64533ac86cf30540f1145553a0418c79bf5#diff-63495596a6a2d9c7803d0a490618a51baa6ab1178c89cdcc3f612f451e76c190R140